test/Microsoft.ML.Tests/TextLoaderTests.cs

```diff
@@ -1002,6 +1002,105 @@ public void TestWrongDecimalMarkerInputs(bool useCommaAsDecimalMarker)
             }
         }
 
+        [Theory]
+        [InlineData(true, true)]
+        [InlineData(false, false)]
+        [InlineData(true, false)]
+        [InlineData(false, true)]
+        public void TestDifferentDecimalMarkersAtTheSameTime(bool useCorrectPeriod, bool useCorrectComma)
+        {
+            // Using 2 different textloaders, with different decimal markers
+            // should yield the expected results even when using their cursors at the same time
+            // in all of the scenarios tested here
```

Comment:
The four scenarios tested here would all fail before this PR, as race conditions would make the cursors to behave and parse the values incorrectly.